index.rb

```diff
@@ -0,0 +1,180 @@
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  def my_each
+    if block_given?
+      0.upto(length - 1) do |i|
+        yield self[i]
+      end
+    else
+      puts 'no block is given'
+    end
+    self
+  end
```

Comment:
`my_each` method should return an `Enumerator` when no block is given. 
Check if a block is given at the beginning of the method if no block is given return an `Enumerator` instead else continue with the method logic.

See the difference below:

```ruby
[1,2,3].each
#=> #<Enumerator: [1, 2, 3]:each>

[1,2,3].my_each
no block is given
#=> [1, 2, 3]
```

HINT: Add `return self.to_enum unless block_given?`

### Let's do this for `my_each_with_index` and `my_select`